src/content/devices/input-output/js/test.js

```diff
@@ -1,143 +1,20 @@
-
-      switch (browser) {
-      case 'chrome':
-        fakeAudioDeviceNames = [
-          'Fake Default Audio Input', // Chrome <= 63
-          'Fake Default Audio Input - Fake Audio Input 1', // Chrome 64+
-          'Fake Audio Input 1',
-          'Fake Audio Input 2'
-        ];
-        break;
-      case 'firefox':
-        fakeAudioDeviceNames = ['Default Audio Device'];
-        break;
-      default:
-        t.skip('unsupported browser');
-      }
-      console.log(fakeAudioDeviceNames, deviceLabel,
-        fakeAudioDeviceNames.indexOf(deviceLabel));
-      t.ok(fakeAudioDeviceNames.indexOf(deviceLabel) !== -1,
-        `Fake audio device found with label: ${deviceLabel}`);
-    })
-    // Check for a fake video device label (Chrome only).
-    .then(() => driver.executeScript('return stream.getVideoTracks()[0].label'))
-    .then(deviceLabel => {
-      let fakeVideoDeviceName = null;
-
-      switch (browser) {
-      case 'chrome':
-        fakeVideoDeviceName = 'fake_device_0';
-        break;
-      case 'firefox':
-        // TODO: Remove the "deviceLabel === ''" check once Firefox ESR
-        // reaches 46 (supports device labels for fake devices).
-        fakeVideoDeviceName = (deviceLabel === '') ? '' : 'Default Video Device';
-        break;
-      default:
-        t.pass('unsupported browser');
-        throw 'skip-test';
-      }
-
-      t.ok(fakeVideoDeviceName === deviceLabel, `Fake video device found with label: ${deviceLabel}`);
-    })
-    // Check that there is a video element and it is displaying something.
-    .then(() => driver.findElement(webdriver.By.id('video')))
-    .then(videoElement => {
-      t.pass('Found video element');
-      let width = 0;
-      let height = 0;
-      return new webdriver.promise.Promise(resolve => {
-        videoElement.getAttribute('videoWidth').then(w => {
-          width = w;
-          t.pass(`Got videoWidth ${w}`);
-          if (width && height) {
-            resolve([width, height]);
-          }
-        });
-        videoElement.getAttribute('videoHeight').then(h => {
-          height = h;
-          t.pass(`Got videoHeight ${h}`);
-          if (width && height) {
-            resolve([width, height]);
-          }
-        });
-      });
-    })
-    .then(dimensions => {
-      t.pass(`Got video dimensions ${dimensions.join('x')}`);
-    })
-    .then(() => {
-      t.end();
-    })
-    .then(null, err => {
-      t.fail(err);
-      t.end();
-    });
-});
+export default {
+  'It should have select elements for each media device': (browser) => {
+    const path = '/src/content/devices/input-output/index.html';
+    const url = 'file://' + process.cwd() + path;
+
+    browser
+      .url(url)
+      .waitForElementVisible('#audioSource option:nth-of-type(1)', 1000, 'Check that there is at least one audio source')
+      .waitForElementVisible('#audioOutput option:nth-of-type(1)', 1000, 'Check that there is at least one audio output')
```

Comment:
this won't pass in Firefox as it does not implement output devices yet.

Reply:
Good to know. Currently fighting with the geckodriver, so I guess I would run into this once that happens. :)